curp/src/client.rs

```diff
@@ -241,8 +241,16 @@ where
                 SyncResult::Error(SyncError::Timeout) => {
                     return Err(ProposeError::SyncedError("wait sync timeout".to_owned()));
                 }
-                SyncResult::Error(e) => {
-                    return Err(ProposeError::SyncedError(format!("{e:?}")));
+                SyncResult::Error(SyncError::NoSuchCmd(id)) => {
```

Comment:
Do you consider refactoring SyncedError just like ExeError in this PR? String is not a good practice for an error type after all.

Reply:
The SyncedError comes from the `SyncError` in curp. So it's a curp's internal error and is like unrecoverable for the client, that a single error type is enough?